packages/salesforcedx-vscode-core/src/commands/forceSandboxClone.ts

```diff
@@ -0,0 +1,235 @@
+    let defaultAlias = DEFAULT_ALIAS;
+    if (hasRootWorkspace()) {
+      const folderName = getRootWorkspace().name.replace(
+        /\W/g /* Replace all non-alphanumeric characters */,
+        ''
+      );
+      defaultAlias = isAlphaNumSpaceString(folderName)
+        ? folderName
+        : DEFAULT_ALIAS;
+    }
+
+    const cloneSourceInputOption = {
+      prompt: nls.localize('parameter_gatherer_enter_source_org'),
+      placeHolder: defaultAlias,
+      validateInput: value => {
+        return isAlphaNumSpaceString(value) || value === ''
+          ? null
+          : nls.localize('error_invalid_org_alias');
+      }
+    } as vscode.InputBoxOptions;
+    const cloneSource = await vscode.window.showInputBox(cloneSourceInputOption);
+    // Hitting enter with no clone source specified, we use the clone source from def file
+    if (cloneSource === undefined) {
+      return { type: 'CANCEL' };
+    }
+
+    const sandboxNameInputOption = {
+      prompt: nls.localize('parameter_gatherer_enter_sandbox_name'),
+      placeHolder: defaultAlias,
+      validateInput: value => {
+        return isAlphaNumSpaceString(value) || value === ''
+          ? null
+          : nls.localize('error_invalid_org_alias');
+      }
+    } as vscode.InputBoxOptions;
+    const sandboxName = await vscode.window.showInputBox(sandboxNameInputOption);
+    if (sandboxName === undefined) {
+      return { type: 'CANCEL' };
+    }
+
+    const aliasInputOptions = {
+      prompt: nls.localize('parameter_gatherer_enter_alias_name'),
+      placeHolder: defaultAlias,
+      validateInput: value => {
+        return isAlphaNumSpaceString(value) || value === ''
+          ? null
+          : nls.localize('error_invalid_org_alias');
+      }
+    } as vscode.InputBoxOptions;
+    const alias = await vscode.window.showInputBox(aliasInputOptions);
+    // Hitting enter with no alias will use the value of `defaultAlias`
+    if (alias === undefined) {
+      return { type: 'CANCEL' };
+    }
+    const waitTimeInMinsInputOptions = {
+      prompt: nls.localize(
+        'parameter_gatherer_enter_sandbox_org_wait_time_mins'
+      ),
+      placeHolder: defaultWaitTime,
+      validateInput: value => {
+        return isIntegerInRange(value, [0, 120]) || value === ''
+          ? null
+          : nls.localize('error_invalid_wait_time_mins');
+      }
+    } as vscode.InputBoxOptions;
+    const scratchOrgExpirationInDays = await vscode.window.showInputBox(
+      waitTimeInMinsInputOptions
+    );
+    if (scratchOrgExpirationInDays === undefined) {
+      return { type: 'CANCEL' };
+    }
+    return {
+      type: 'CONTINUE',
+      data: {
+        alias: alias === '' ? defaultAlias : alias,
+        waitTime:
+          scratchOrgExpirationInDays === ''
+            ? defaultWaitTime
+            : scratchOrgExpirationInDays,
+        sourceSandbox: cloneSource,
+        sandboxName
+      }
+    };
+  }
+}
+export interface Alias {
```

Comment:
This could be moved somewhere else where other commands using aliases can use it. Maybe that means having a simplified version of it so it can be extended when needed.